procmonAnalyze.js

```diff
@@ -334,65 +356,102 @@ async function drawData(data, diskify) {
   let totalTime = maxTime - minTime;
   let trackWidth = canvas.width / tracks.length;
   let rendererScale = canvas.height / totalTime;
+
+  if (!isRefocused) {
+    gOuterState = {
+      minTime,
+      maxTime,
+      totalTime,
+      unconstrainedData: aData,
+      unconstrainedDiskifyData: aData,
+    };
+  }
+
   gState = {
-    minTime,
-    maxTime,
-    tracks,
-    totalTime,
-    trackWidth,
-    rendererScale,
-    targetRendererScale: rendererScale,
-    diskmapScale,
-    targetDiskmapScale: diskmapScale,
-    readsByPath,
-    diskify,
-    maxLcn,
-    diskmapTranslate: 0,
-    targetDiskmapTranslate: 0,
-    rendererTranslate: 0,
-    targetRendererTranslate: 0,
-    mouseX: 0,
-    mouseY: 0,
-    timelineIndicators: [],
-    lcnReads: [],
-    lastHoveredRect: null,
-    selectedEntry: null,
-    activePath: null,
+      minTime,
+      maxTime,
+      tracks,
+      totalTime,
+      trackWidth,
+      rendererScale,
+      targetRendererScale: rendererScale,
+      diskmapScale,
+      targetDiskmapScale: diskmapScale,
+      readsByPath,
+      diskify,
+      maxLcn,
+      diskmapTranslate: 0,
+      targetDiskmapTranslate: 0,
+      rendererTranslate: 0,
+      targetRendererTranslate: 0,
+      mouseX: 0,
+      mouseY: 0,
+      timelineIndicators: [],
+      lcnReads: [],
+      lastHoveredRect: null,
+      selectedEntry: null,
+      activePath: null,
+      topMarkerCoordinate: 0,
+      bottomMarkerCoordinate: markersCanvas.height,
+      isRefocused: isRefocused,
+      minTimeRelative: absTimeToRelTime(minTime),
+      maxTimeRelative: absTimeToRelTime(maxTime),
   };
 
+  gOuterState.constrainedTimeBegin = minTime;
```

Comment:
Let's move this outside of `drawData` calls. I think it would be preferable if this function does not modify `gOuterState`, and as much as possible restricts itself to thinking about only `gState`.